### DISCUSSION TITLE
Heat Pump Error

### DISCUSSION THREAD

**Discussion by Crewski** [tagyoureit/nodejs-poolController#449] [Nixie] [answered]
I'm new to njsPC so forgive me if I'm missing something obvious.  I am trying to setup my heat pump with a fireman switch.  I'm currently bench testing my unit.  I have the fireman switch setup to a relay that is on GPIO27.  GPIO27 is setup in REM and works when I click the button in REM.  The heat pump is setup with the heater relay in dashPanel.  However, I can't seem to get the heat pump to turn on.  The logs have an error that says 

```
Error synchronizing heater states: Cannot read property 'getTime' of undefined
```

Any ideas?  I'm on release 7.7, just installed it yesterday.  Thanks for the help.

For testing, if I put the heater as being a gas heater, the relay turn on/off as expected.  Switching back to heat pump causes it to stop working and get the same errors about getTime.

![image](https://user-images.githubusercontent.com/8012153/169184920-4814412e-7c14-470e-9869-9ec86240806f.png)
![image](https://user-images.githubusercontent.com/8012153/169184969-7affa418-0421-4ce7-9e29-a8038c0d54da.png)
![image](https://user-images.githubusercontent.com/8012153/169185028-bba94ae3-3e29-4ab0-ae0f-616610cd66f4.png)

**Comment by rstrouse** [comment]
Can I get you to produce a replay when you engage the heater and upload it here?  Also which heat pump do you have?

**Reply by Crewski** [reply]
I'm not sure what you mean by produce a replay, can you give some direction on that?  I have a Hayward Heat Pump that only has the "2-wire" control for external controller.  Not sure if you replied before my edit, but if I set the heater as a gas heater the relay works as expected.

Right now I'm just doing some bench testing before actual install, so it isn't actually hooked up to the heater.  I have ds18b20 probes providing the temperatures.

**Reply by rstrouse** [reply]
The instructions are here. https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution

**Reply by Crewski** [reply]
Man you guys made it easy to capture stuff.  Attached is the replay.
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/8723034/replay.zip)
.

**Comment by rstrouse** [comment]
Pull njsPC it should now work with a 2 wire heatpump hookup.

**Reply by Crewski** [reply]
Its seems to be working now, except it says that it is cooling instead of heating.
![image](https://user-images.githubusercontent.com/8012153/169386999-88c0af13-f400-4f61-b019-654e069bffdb.png)

EDIT:  It changed to heating.  I restarted njsPC again it it starts off as showing cooling, but then switches over to colling after  few seconds.

**Comment by rstrouse** [comment, marked as answer]
Pull njsPC.  The cooling function needed to be disabled for 2-wire hookups on the heatpump.

**Reply by Crewski** [reply]
Everything seems to be working  now and the relay is probably triggering.  Thanks for the support!

**Comment by Crewski** [comment]
The heat pump is operating properly, but it doesn't seem to be trigging the "pool heater" circuit...so I can't seem to have my pump speed up when the heater is turned on.  The "any heater" circuit turns on, but for whatever reason when I select that for the pump, it doesn't display any text or set the speed.
![image](https://user-images.githubusercontent.com/8012153/171073381-5d16135f-3ab0-41df-9cd5-2450d3fbb71a.png)

[replay (11).zip](https://github.com/tagyoureit/nodejs-poolController/files/8801214/replay.11.zip)

**Reply by rstrouse** [reply]
Hmm... Do you have solar?  You shouldn't have a solar virtual circuit if there is no solar heater.  I will take a look at your replay.

**Comment by Crewski** [comment]
No solar. I do have a temp sensor on the output of my heater that is set as
watersensor2. I used it just to confirm heater operation.

**Reply by rstrouse** [reply]
Pull njsPC the heatpump states were not included in the heater virtual circuits.
